Marketplace ContentAPI changes

serdarceranmendix opened the pull request:
Frst get approval from internal stakeholders (@KseniiaIankina)
DONE

serdarceranmendix committed:
Create content-api for marketplace

serdarceranmendix committed:
Rename content-api to content-api.md

serdarceranmendix committed:
Create marketplace-content.yaml

serdarceranmendix committed:
Update supportCategory in marketplace-content.yaml

serdarceranmendix committed:
Update the pattern for the version in marketplace-content.yaml

serdarceranmendix committed:
Update in the version patterns in marketplace-content.yaml

serdarceranmendix committed:
Update isPrivate to string in marketplace-content.yaml

katarzyna-koltun-mx commented:
@Luyao-Zhang-1 let me know if the PR should be reassigned.

serdarceranmendix committed:
Add required part for specificContent in marketplace-content.yaml

serdarceranmendix committed:
Update content-api.md with additional information.

serdarceranmendix committed:
Updated the additional information in  content-api.md

serdarceranmendix committed:
Added alert for beta program in content-api.md

serdarceranmendix committed:
Added contact info for beta program in blue section in content-api.md

serdarceranmendix committed:
underline the contact info in content-api.md

serdarceranmendix committed:
Removed the email part from contact section in marketplace-content.yaml

serdarceranmendix committed:
Update content-api.md

serdarceranmendix committed:
Update content-api.md

serdarceranmendix committed:
Update content-api.md

serdarceranmendix committed:
Update content-api.md

serdarceranmendix committed:
Update content-api.md

ConnorLand commented on content/en/docs/apidocs-mxsdk/apidocs/content-api.md:
```diff
@@ -0,0 +1,51 @@
+---
+
+{{% alert color="info" %}}
+This API is only available via a private beta program. If you want to be part of this program, kindly reach out to [<u>Marketplace team</u>](mailto:Marketplace_Team@Mendix.com) to get access.
```
@Luyao-Zhang-1 I have tested in a local build, and these underline divs do work successfully. However, I think underline styling is not needed. The regular way our CSS makes hyperlink text pop will draw attention to the link quite well.

serdarceranmendix replied:
Hi @ConnorLand ,
No, underline styling on "Marketplace team" email link is needed indeed.

> underline styling is not needed

KseniiaIankina replied:
Hi all, @ConnorLand if it puts attendtion that it is a link, then great, we do not need it underlined. Thank you

serdarceranmendix committed:
Update content-api.md

serdarceranmendix committed:
Update content-api.md

serdarceranmendix committed:
Update the descriptions in marketplace-content.yaml

serdarceranmendix committed:
The info section for the beta program is removed in content-api.md

serdarceranmendix commented:
Hi @katarzyna-koltun-mx and @Luyao-Zhang-1,
As we are still in the beta program phase, we have decided to postpone the release of the documentation by one month. We believe it's essential to refine our API based on the feedback that we will receive during the beta program before we finalize and publish the documentation.

Is it also fine for you?

Adam-Dupaski commented:
Hi @serdarceranmendix yes, if the API is currently only in Private Beta, we can wait to publish until it is Public Beta or GA, that is fine.

Commit:
Update content-api.md

Commit:
Update content-api.md

Luyao-Zhang-1 committed:
Update values of weight

Luyao-Zhang-1 committed:
fix typo

Luyao-Zhang-1 committed:
Update marketplace-content.yaml

Luyao-Zhang-1 committed:
Update content-api.md

Luyao-Zhang-1 committed:
Update marketplace-content.yaml

serdarceranmendix committed:
Updated description of "/content" API in marketplace-content.yaml

Added: "The content metadata is from the last published version of the content."

Luyao-Zhang-1 commented on static/openapi-spec/marketplace-content.yaml:
```diff
@@ -0,0 +1,382 @@
+          description: "This indicates whether the content is public or private."
+        latestVersion:
+          $ref: '#/components/schemas/version'
+    content:
```
@serdarceranmendix  Do you think we should add "description" to "content"?

serdarceranmendix replied:
I just added for 'content'. can you please check @Luyao-Zhang-1 ?

Luyao-Zhang-1 commented on static/openapi-spec/marketplace-content.yaml:
```diff
@@ -0,0 +1,382 @@
+          type: array
+          items:
+            $ref: '#/components/schemas/specificContent'
+    contentVersions:
```
@serdarceranmendix Do you think we should add "description" to "ContentVersions"?

serdarceranmendix replied:
The description is not needed for this one. @Luyao-Zhang-1 .

serdarceranmendix committed:
Added description for content in marketplace-content.yaml

Luyao-Zhang-1 committed:
Update marketplace-content.yaml

Luyao-Zhang-1 committed:
Merge branch 'development' into pr/6649

Luyao-Zhang-1 committed:
Merge branch 'development' of https://github.com/serdarceranmendix/docs into pr/6649
